library/ssl_tls.c

```diff
@@ -11363,17 +11378,19 @@ int mbedtls_ssl_write( mbedtls_ssl_context *ssl, const unsigned char *buf, size_
 #if defined(MBEDTLS_SSL_CBC_RECORD_SPLITTING)
     ret = ssl_write_split( ssl, buf, len );
 #else
+#if defined(MBEDTLS_FI_COUNTERMEASURES)
+    /*Add const user pointers to context. We will be able to check its validity before copy to context*/
+    ssl->out_msg_dup = (unsigned char*)buf_dup;
```

Comment:
`buf_dup` will be defined only if `MBEDTLS_SSL_CBC_RECORD_SPLITTING` is not defined, and in this line, it only depends on `MBEDTLS_FI_COUNTERMEASURES`.